### DISCUSSION TITLE
De/CompressData usage

### DISCUSSION THREAD

**Discussion by wereii** [raysan5/raylib#1795] [Q&A] [answered]
Hi, wanted to ask someone more knowledgeable about raylib (was looking for IRC:)

I can't seem to get `DecompressData` to work. Is this just in-house custom compression? eg. is it expected that the compressed data were passed thru `CompressData` or is it generic DEFLATE ?

Because, trying to decompress either zlib or gzip (I know, gzip has headers on top, just tried) fails, probably trying to allocate too much memory (see picture). Was trying to decompress file created with `echo "Hello World" | zlib-flate -compress > hi.z`
![image](https://user-images.githubusercontent.com/8504482/120073876-8a235500-c09a-11eb-8a88-7d0bc8642686.png)

Also, this error is basically un-catchable outside because there is no way to know that it actually failed somewhere.
What am I doing wrong here?

https://gist.github.com/wereii/6711de7e21667f0cf6d19e685524b560

**Comment by raysan5** [comment, marked as answer]
Hi @wereii, 

raylib compression/decompression functions use internally the libraries [`sdefl.h`](https://github.com/vurtun/lib/blob/master/sdefl.h) and [`sinfl.h`](https://github.com/vurtun/lib/blob/master/sinfl.h) that implement the standard Deflate (RFC 1951) algorithm, so, they should work with any stream using that same algorithm.

I just tested it with PNGs zlib-streams (Deflate with pre-filtering on scanlines) and it worked ok with any PNG. Are you sure that those methods you used generate a pure Deflate (RFC 1951) stream? Maybe some header data is added automatically...

**Reply by wereii** [reply]
You are most probably right that my way of generating deflate data adds various headers around.
Currently working this out, once I verify this in-code I will mark your comment as an answer.
